src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -204,14 +209,61 @@ private static SuppressionDiffResult CompareFiles(SuppressionFile oldFile, Suppr
 		{
 			cancellation.ThrowIfCancellationRequested();
 
-			if (settings.SuppressionMechanismEnabled && Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true)
+			if (settings.SuppressionMechanismEnabled && 
+			    (Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true ||
+			    CheckSuppressedComment(diagnostic, cancellation)))
 			{
 				return;
 			}
 
 			reportDiagnostic(diagnostic);
 		}
 
+		private static bool CheckSuppressedComment(Diagnostic diagnostic, CancellationToken cancellation)
+		{
+			cancellation.ThrowIfCancellationRequested();
+			SyntaxNode root = diagnostic.Location.SourceTree.GetRoot();
+			SyntaxNode node;
+
+			bool containsComment = false;
+
+			// Climb to the hill. Looking for comment on parents nodes.
+			for (node = root.FindNode(diagnostic.Location.SourceSpan); node != root ; node = node.Parent)
+			{
+				containsComment = CheckSuppressionCommentOnNode(diagnostic, node, cancellation);
+				
+				if (node is StatementSyntax || node is MemberDeclarationSyntax || containsComment)
+					break;
+			}
+
+			return containsComment;
+		}
+
+		private static bool CheckSuppressionCommentOnNode(Diagnostic diagnostic, SyntaxNode node, CancellationToken cancellation)
+		{
+			cancellation.ThrowIfCancellationRequested();
+
+			if (node != null && node.HasLeadingTrivia && node.GetLeadingTrivia().Any(SyntaxKind.SingleLineCommentTrivia))
+			{
+				var comments = node.GetLeadingTrivia()
+					.Where(x => x.RawKind == (int)SyntaxKind.SingleLineCommentTrivia);
+
+				foreach (SyntaxTrivia comment in comments)
+				{
+					Match match = _suppressPattern.Match(comment.ToString());
+
+					if (match.Success &&
+					    string.Equals(diagnostic.Id, match.Groups[1].Value, StringComparison.OrdinalIgnoreCase) && 
+					    string.Equals(diagnostic.Descriptor.CustomTags.FirstOrDefault(), match.Groups[2].Value, StringComparison.OrdinalIgnoreCase))
```

Comment:
Retrieving diagnostic short name from CustomTags for each node on each iteration is not a good thing for performance. 
It can be passed to the method. See the previous comment for an example. 
Comparison of string with the specified culture sometimes clutters the code. I know that is is adviced to always specify string comparison, however in this case imho the simple `==` is much shorter and more readable.